tests/progs/test_random_refinement.py

```diff
@@ -1,18 +1,23 @@
-from progs.random_refinement import refine_randoms
-from honeybadgermpc.polynomial import get_omega
 from pytest import mark
+from honeybadgermpc.mpc import TaskProgramRunner
+from progs.random_refinement import refine_randoms
 
 
 @mark.parametrize("n, t, k", [(4, 1, 3), (4, 1, 4), (7, 2, 5)])
-def test_random_refinement(n, t, k, galois_field, test_preprocessing, polynomial):
-    random_shares_gf = [galois_field.random() for _ in range(k)]
-    random_shares_int = [int(i) for i in random_shares_gf]
-    d = 2**(k-1).bit_length()
-    omega = get_omega(galois_field, 2*d, seed=0)
-    xs = [pow(omega, 2*i) for i in range(d)]
-    poly = polynomial.interpolate(zip(xs, random_shares_gf))
-    refined_share_gfs = refine_randoms(n, t, galois_field, random_shares_int)
-    assert len(refined_share_gfs) == k-t
-    for i, share in enumerate(refined_share_gfs):
-        assert isinstance(share, int)
-        poly(pow(omega, 2*i+1)) == share
+@mark.asyncio
+async def test_random_refinement(n, t, k, galois_field, test_preprocessing, polynomial):
+    test_preprocessing.generate("rands", n, t)
+
+    async def _prog(context):
+        random_shares = [test_preprocessing.elements.get_rand(
+            context).v.value for i in range(k)]
+        refined_random_shares = refine_randoms(n, t, galois_field, random_shares)
+        assert len(refined_random_shares) == k-t
+        randoms = await context.ShareArray(refined_random_shares).open()
+        return tuple(randoms)
+
+    program_runner = TaskProgramRunner(n, t)
```

Comment:
BTW, you can cut out this boilerplate with the `test_runner` fixture.

Reply:
Done!